src/main/java/com/uvocab/api/service/AuthService.java

```diff
@@ -0,0 +1,29 @@
+package com.uvocab.api.service;
+
+import lombok.AllArgsConstructor;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import uvocab.protobuf.v1.User;
+
+@Service
+@AllArgsConstructor
+public class AuthService {
+
+  private final AuthenticationManager authenticationManager;
+
+  private static final String FIXED_TOKEN = "str_token";
+
+  public String login(User user) {
+    Authentication authentication =
+        authenticationManager.authenticate(
+            new UsernamePasswordAuthenticationToken(user.getLogin(), user.getPassword()));
+    SecurityContextHolder.getContext().setAuthentication(authentication);
+    if (authentication.isAuthenticated()){
+        return FIXED_TOKEN;
+    }
+    else throw new RuntimeException("Falha ao autenticar");
+  }
```

Comment:
as I said in comment above :arrow_up: , the code is too much, because when call the **authenticate**, this method will do all we need.... get the **SecurityContextHolder**, check is authenticated **(authentication.isAuthenticated**) and throw a Exception no make sense.

p.s: throw a exception here no make sense and one more **RuntimeException** , Know the concepts about checked and not checked exceptions in Java is crucial to understand when use It.